src/huggingface_hub/repository.py

```diff
@@ -239,7 +239,7 @@ def is_binary_file(filename: Union[str, Path]) -> bool:
     """
     try:
         with open(filename) as f:
-            content = f.read()
+            content = f.read(512)
```

Comment:
512 bytes is a low number for the heuristic of only checking for the null character (on the API the heuristic is more complex).

If bytes are uniformly distributed, you will have 13% failure (used [this calculator](https://www.omnicalculator.com/statistics/probability)).

Also, with reading 512 bytes, I'm not sure if it can create a `UnicodeDecodeError` in a text file if cutting in the middle of a character at the end? I don't know enough about python to answer that...

Reading the whole file is ok with me, but in the `auto_track_binary_files` function, the `is_binary` check should be done _after_ checking whether the file is already tracked by LFS or in the gitignore.

That way, we'd ensure that at most 10MB are loaded in memory, since otherwise the `auto_track_large_files` would have already marked it LFS.

Reply:
Good feedback, thanks @coyotte508. Will adapt.

Reply:
yes good future proofing :)